### PR TITLE
Update number.blade.php

### PR DESCRIPTION
**Pull request by abarta**
Change needed to allow empty values in old() to clear a field on page reload. Needs to be done in all field. Have done it only for text and number fields.

### CODE REVIEW

**Comment by tabacitu** [comment]
Thank you @abarta !

File: src/resources/views/fields/number.blade.php [hunk 1]
```diff
@@ -9,7 +9,7 @@
         	type="number"
         	name="{{ $field['name'] }}"
             id="{{ $field['name'] }}"
-            value="{{ old($field['name']) ? old($field['name']) : (isset($field['value']) ? $field['value'] : (isset($field['default']) ? $field['default'] : '' )) }}"
+            value="{{ old($field['name']) !== null ? old($field['name']) : (isset($field['value']) ? $field['value'] : (isset($field['default']) ? $field['default'] : '' )) }}"
```
Comment (lloy0076) [inline src/resources/views/fields/number.blade.php]: `!is_null(...)` seems more elegant unless I'm missing something here.

In fact that's such a ternary operator mess that I'd be inclined to refactor this some other way.

Comment (tabacitu) [reply src/resources/views/fields/number.blade.php]: Totally agree with you here @lloy0076 . Also agree with @abarta that this is a problem. So I have something very nice for you guys :-) Null coalesce operator will fix both your issues :-)

So:
```diff
-{{ old($field['name']) !== null ? old($field['name']) : (isset($field['value']) ? $field['value'] : (isset($field['default']) ? $field['default'] : '' )) }}
+{{ old($field['name']) ?? $field['value'] ?? $field['default'] ?? '' }}
```

Pretty sexy, right? :-)

Comment (tabacitu) [reply src/resources/views/fields/number.blade.php]: Gotta love PHP 7 for this :-)

**Comment by tabacitu** [comment]
@abarta , @lloy0076 , thanks for the help. Just merged this into another branch and done the same for all fields. Check it out here - https://github.com/Laravel-Backpack/CRUD/pull/1690

@abarta - thanks again.

Cheers!

**Comment by zachweix** [comment]
@tabacitu - random question, I see that you updated a lot of fields and converted them all to use the double question mark null coalescence operator? Was that an outgrowth of its use in the select.blade.php field?

**Pull request closed without merging**